src/dos/drive_cache.cpp

```diff
@@ -417,31 +411,34 @@ int DOS_Drive_Cache::CompareShortname(const char* compareName, const char* short
 	return strcmp(compareName,shortName);
 }
 
-Bitu DOS_Drive_Cache::CreateShortNameID(CFileInfo* curDir, const char* name) {
-	std::vector<CFileInfo*>::size_type filelist_size = curDir->longNameList.size();
-	if (GCC_UNLIKELY(filelist_size<=0)) return 1;	// shortener IDs start with 1
-
-	Bitu foundNr	= 0;	
-	Bits low		= 0;
-	Bits high		= (Bits)(filelist_size-1);
-	Bits mid, res;
-
-	while (low<=high) {
-		mid = (low+high)/2;
-		res = CompareShortname(name,curDir->longNameList[mid]->shortname);
+unsigned DOS_Drive_Cache::CreateShortNameID(CFileInfo *curDir, const char *name)
+{
+	assert(curDir);
+	const auto filelist_size = curDir->longNameList.size();
```

Comment:
Nice use of 'auto' to accommodate the possible different return types and sizes across architectures, C++ versions, and any bugs / differences they might have.

Reply:
Thanks :)